tests/e2e/files/rulebooks/operators/test_selectattr_operator.yml

```diff
@@ -0,0 +1,200 @@
+              - asteroid:
+                  name: "vesta"
+                  category: "main belt"
+                  description: |
+                    Named after the goddess of
+                    the hearth and household in
+                    Roman mythology.
+          - id: "testcase negative"
+            asteroids:
+              - asteroid:
+                  name: "eros"
+                  discovered_by: "gustav witt"
+                  radius: 8.42
+          - id: "testcase 03"
+            planets:
+              - planet:
+                  name: "jupiter"
+                  radius: 142984
+                  orbital_period: 4311.86
+          - id: "testcase 03"
+            planets:
+              - planet:
+                  name: "earth"
+                  radius: 6371
+                  orbital_period: 365.01
+          - id: "testcase 04"
+            planets:
+              - planet:
+                  name: "venus"
+                  radius: 1200.01
+                  moons: null
+                  is_planet: true
+              - planet:
+                  name: "x"
+                  radius: 909.09
+                  moons: null
+                  is_planet: false
+          - id: "testcase 05"
+            planets:
+              - planet:
+                  name: "pluto"
+                  radius: 1188.3
+                  total_moons: 5
+                  is_planet: false
+          - id: "testcase 06"
+            planets:
+              - planet:
+                  name: "mars"
+                  moons:
+                    - "phobos"
+                    - "deimos"
+              - planet:
+                  name: "earth"
+                  moons:
+                    - "moon"
+          - id: "testcase 07"
+            planets:
+              - planet:
+                  name: "eris"
+                  moons:
+                    - "dysnomia"
+          - id: "testcase 08"
+            planets:
+              - planet:
+                  name: "saturn"
+                  total_moons: 62
+                  largest_moon: "titan"
+          - id: "testcase 09"
+            planets:
+              - planet:
+                  name: "neptune"
+                  total_moons: 14
+                  largest_moon: "triton"
+              - planet:
+                  name: "uranus"
+                  total_moons: 27
+                  largest_moon: "titania"
+          - id: "testcase 09"
+            planets:
+              - planet:
+                  name: "earth"
+                  total_moons: 1
+                  largest_moon: "moon"
+          - id: "testcase 10"
+            suns:
+              - sun:
+                  name: "sol"
+                  attributes:
+                    physical:
+                      age: 4600000000
+                      composition:
+                        gases:
+                          hydrogen: 73.46
+                          helium: 24.85
+
+  rules:
```

Comment:
We need at least one usecase with variable substitution, don't?

Reply:
@Alex-Izquierdo thankyou for raising this, there should have been a note in each file pointing to the bug that needs to be resolved for this to work: https://issues.redhat.com/browse/AAP-9324

I've added a comment to both files.